addons/point_of_sale/static/src/app/store/models.js

```diff
@@ -947,16 +949,63 @@ export class Payment extends PosModel {
         return Boolean(this.get_payment_status());
     }
 
+    async showQR() {
+        let qr;
+        try {
+            qr = await this.env.services.orm.call("pos.payment.method", "get_qr_code", [
+                [this.payment_method.id],
+                this.amount,
+                this.order.name,
+                this.order.name,
+                this.pos.currency.id,
+                this.order.partner?.id,
+            ]);
+        } catch (error) {
+            qr = this.payment_method.default_qr;
+            if (!qr) {
+                let message;
+                if (error instanceof ConnectionLostError) {
+                    message = _t(
+                        "Connection to the server has been lost. Please check your internet connection."
+                    );
+                } else {
+                    message = error.data.message;
+                }
+                this.env.services.dialog.add(AlertDialog, {
+                    title: _t("Failure to generate Payment QR Code"),
+                    body: message,
+                });
+                return false;
+            }
+        }
+        return await ask(
+            this.env.services.dialog,
+            {
+                title: _t(this.name),
+                line: this,
+                order: this.order,
+                qrCode: qr,
+            },
+            {},
+            QRPopup
+        );
+    }
+
     async pay() {
         this.set_payment_status("waiting");
+        if (this.payment_method.payment_method_type === "qr_code") {
+            return this.handle_payment_response(await this.showQR());
+        }
         return this.handle_payment_response(
             await this.payment_method.payment_terminal.send_payment_request(this.cid)
         );
     }
     handle_payment_response(isPaymentSuccessful) {
         if (isPaymentSuccessful) {
             this.set_payment_status("done");
-            this.can_be_reversed = this.payment_method.payment_terminal.supports_reversals;
+            if (!this.payment_method.payment_method_type === "qr_code") {
```

Comment:
```suggestion
            if (this.payment_method.payment_method_type !== "qr_code") {
```
is this what you intended?

Reply:
Not sure how I ended up writing that lol